### ISSUE TITLE
[J10] Prepare Java 10 Support

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext#1177]
With the Photon Release we should be able to support java 10

- `JavaVersion` constant and its from and to conversions => IDEA API we use does not support J10 but i wont care at this point
- Wizard
- new Maven/Gradle Dependencies to JDT we consume needed, poblishing probably with oxygen 3a in beginning of april (https://dev.eclipse.org/mhonarc/lists/jdt-dev/msg00978.html)
- ASM (i think asm 6.0 would be sufficient)? But we should at least check if we create a range [6.0,6.2) in gradle - created (http://dev.eclipse.org/ipzilla/show_bug.cgi?id=16102) for 6.1 and possibly we need even 6.1.1 (https://bugs.eclipse.org/bugs/show_bug.cgi?id=532291) (https://dev.eclipse.org/ipzilla/show_bug.cgi?id=16105)
- new Tycho Version needed https://bugs.eclipse.org/bugs/show_bug.cgi?id=532233
- Gradle? (could be we need gradlew 4.6 and thus an new cq. i created https://dev.eclipse.org/ipzilla/show_bug.cgi?id=15868 :white_check_mark:) - wizard need to be adapted :white_check_mark:
- Maven? I assume will work ootb
- Target Platform in Wizard
- Function wise (do not explode)

**Comment by cdietrich** [comment]
we should keep https://github.com/eclipse/xtext-core/issues/517 in mind while doing this

**Comment by cdietrich** [comment]
Oxygen 3a seems not to be on photon. is that correct @kthoms ?

**Comment by cdietrich** [comment]
cc @kthoms @arnedeutsch

**Comment by ArneDeutsch** [comment]
https://github.com/ArneDeutsch/xtext-eclipse/commit/2beaf8ee7c1ba0c431988d5cf7c3f1f48bc19c98
depends on
https://github.com/ArneDeutsch/xtext-core/commit/ec683f723ce778c54f2574b89c563b94d0d32bfd
but only the former is merged

**Comment by kthoms** [comment]
Observed a problem building Greeting DSL with Maven on Java 10: Xtend plugin fails compiling MyDslIdeSetup
```
Type mismatch: cannot convert from MyDslIdeModule to Module
```

**Comment by cdietrich** [comment]
please share a link

**Comment by cdietrich** [comment]
TODO: update to TYCHO 1.2.0 release as soon as available.
use stable orbit update site for asm and co as soon as available

**Comment by ArneDeutsch** [comment]
Checked this morning (23.04.2018, 7:45). No tycho 1.2.0 yet, no stable orbit update site with asm 6.1.1 yet.

**Comment by cdietrich** [comment]
yes. not even the snapshot has full java 10 capabilies. ntl we will ship with tycho 1.2.0 snapshot configured in the beta so that one they ship that it will be a drop in replacement.

this is why i left this ticket open so that we can update to 1.2.0 final once its out and forget to change the orbit site to a stable once photon M7 is out

**Comment by cdietrich** [comment]
Fixed in 2.14
